File: app/src/main/java/dev/ragnarok/fenrir/util/existfile/AbsFileExist.java
package dev.ragnarok.fenrir.util.existfile;

import android.content.Context;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import java.io.IOException;
import java.util.List;

import dev.ragnarok.fenrir.model.wrappers.SelectablePhotoWrapper;
import io.reactivex.rxjava3.core.Completable;

public interface AbsFileExist {
    void findRemoteAudios(Context context) throws IOException;

    Completable findLocalImages(@Nullable List<SelectablePhotoWrapper> photos);

    void addAudio(@NonNull String file);

    Completable findAllAudios(Context context);

    void markExistPhotos(@NonNull List<SelectablePhotoWrapper> photos);

    boolean isExistRemoteAudio(@NonNull String file);

    int isExistAllAudio(@NonNull String file);
}

File: app_fenrir/src/main/kotlin/dev/ragnarok/fenrir/util/existfile/AbsFileExist.kt
package dev.ragnarok.fenrir.util.existfile

import android.content.Context
import dev.ragnarok.fenrir.model.wrappers.SelectablePhotoWrapper
import io.reactivex.rxjava3.core.Completable
import java.io.IOException

interface AbsFileExist {
    @Throws(IOException::class)
    fun findRemoteAudios(context: Context)
    fun findLocalImages(photos: List<SelectablePhotoWrapper>): Completable
    fun addAudio(file: String)
    fun addPhoto(file: String)
    fun findAllAudios(context: Context): Completable
    fun markExistPhotos(photos: List<SelectablePhotoWrapper>)
    fun isExistRemoteAudio(file: String): Boolean
    fun isExistAllAudio(file: String): Int
}